testing/test_terminal.py

```diff
@@ -2614,3 +2614,123 @@ def test_format_trimmed() -> None:
 
     assert _format_trimmed(" ({}) ", msg, len(msg) + 4) == " (unconditional skip) "
     assert _format_trimmed(" ({}) ", msg, len(msg) + 3) == " (unconditional ...) "
+
+
+def test_fine_grained_test_case_verbosity(pytester: Pytester):
```

Comment:
Up to you, but when I have "groups" of tests like this I prefer to group them explicitly using a class, so beside being explicit we can use shorter names for the tests:

```python
def test_fine_grained_test_case_verbosity(pytester: Pytester): ...
def test_fine_grained_test_case_verbosity_collect_only_negative_2(pytester: Pytester): ...
def test_fine_grained_test_case_verbosity_collect_only_positive_2(pytester: Pytester): ...
```

vs:

```python
class TestFineGrainedTestCase:
    def test_verbosity(self, pytester: Pytester): ...
    def test_collect_only_negative_2(self, pytester: Pytester): ...
    def test_collect_only_positive_2(self, pytester: Pytester): ...
```

As a bonus the `_fine_grained_verbosity_file_contents` utility can be turned into a method of the class, making it more context, and can be renamed too to something simpler like `file_contents()` (no need for the `_`).